src/components/picker/picker.tsx

```diff
@@ -252,94 +248,70 @@ export class Picker {
+            return this.renderPortal(false);
         }
 
-        const boundingRect = this.element.getBoundingClientRect();
-
+        const cssList = ['mdc-elevation-transition', 'mdc-elevation--z4'];
         if (this.loading) {
-            return (
-                <div
-                    style={{
-                        width: `${boundingRect.width}px`,
-                    }}
-                    class={`
-                        dropdown--spinner
-                        mdc-elevation-transition
-                        mdc-elevation--z4
-                        mdc-menu-surface
-                        mdc-menu-surface--open
-                    `}
-                    tabindex="-1"
-                >
-                    <limel-spinner />
-                </div>
-            );
+            const dropdownSpinnerStyle = 'dropdown--spinner';
+            cssList.push(dropdownSpinnerStyle);
+            const spinner = <limel-spinner class={dropdownSpinnerStyle} />;
+            return this.renderPortal(true, spinner, cssList);
         }
 
         if (!this.items || !this.items.length) {
             if (!this.emptyResultMessage) {
-                return;
+                return this.renderPortal(false);
             }
-
-            return (
-                <div
+            cssList.push('dropdown--list');
+            const emptyMessage = (
+                <p
                     style={{
-                        width: `${boundingRect.width}px`,
+                        color: 'var(--lime-light-grey, #{$lime-light-grey})',
+                        'text-align': 'center',
                     }}
-                    class={`
-                        dropdown--list
-                        mdc-elevation-transition
-                        mdc-elevation--z4
-                        mdc-menu-surface
-                        mdc-menu-surface--open
-                    `}
-                    tabindex="-1"
                 >
-                    <p class="empty-result-message">
-                        {this.emptyResultMessage}
-                    </p>
-                </div>
+                    {this.emptyResultMessage}
+                </p>
             );
+            return this.renderPortal(true, emptyMessage, cssList);
         }
 
         const hasIcons = this.items.some(item => {
             return 'icon' in item && !!item.icon;
         });
+        const limelList = (
+            <limel-list
+                badgeIcons={hasIcons}
+                onChange={this.handleListChange}
+                type="selectable"
+                items={this.items}
+            />
+        );
+
+        cssList.push('dropdown--list');
+        if (this.displayFullList || this.items.length <= ITEM_LIMIT_NO_SCROLL) {
+            cssList.push('display-full-list');
+        }
+        return this.renderPortal(true, limelList, cssList);
+    }
 
+    private renderPortal(isVisible, content = null, cssList: string[] = []) {
```

Comment:
I don't think we need the `isVisible` argument, it seems we always call this in one of two ways, either
```typescript
this.renderPortal(false)
```
or
```typescript
this.renderPortal(true, content)
```
so we should be able to just check if the `content` argument is set or not.
Also, `this.renderPortal(false)` makes it sound like we are not rendering a portal, so it's a bit confusing :slightly_smiling_face: